src/sentry/search/events/filter.py

```diff
@@ -394,6 +397,56 @@ def parse_semver_search(
     return ["release", final_operator, versions]
 
 
+def parse_semver(version, operator) -> Optional[SemverFilter]:
+    """
+    Attempts to parse a release version using our semver syntax. version should be in
+    format `<package_name>@<version>` or `<version>`, where package_name is a string and
+    version is a version string matching semver format (https://semver.org/). We've
+    slightly extended this format to allow up to 4 integers. EG
+     - sentry@1.2.3.4
+     - sentry@1.2.3.4-alpha
+     - 1.2.3.4
+     - 1.2.3.4-alpha
+     - 1.*
+    """
+    operator = OPERATOR_TO_DJANGO[operator]
+    version = version if "@" in version else f"{SEMVER_FAKE_PACKAGE}@{version}"
+    parsed = parse_release_relay(version)
+    parsed_version = parsed.get("version_parsed")
+    if parsed_version:
+        # Convert `pre` to always be a string
+        prerelease = parsed_version["pre"] if parsed_version["pre"] else ""
+        semver_filter = SemverFilter(
+            operator,
+            [
+                parsed_version["major"],
+                parsed_version["minor"],
+                parsed_version["patch"],
+                parsed_version["revision"],
+                0 if prerelease else 1,
+                prerelease,
+            ],
+        )
+        if parsed["package"] and parsed["package"] != SEMVER_FAKE_PACKAGE:
+            semver_filter.package = parsed.package
```

Comment:
Just realised I missed including the package as part of wildcards, will add a test and fix

Reply:
Actually, I'll do this as a follow-up so that this is easier to review. This function was just moved as-is here, without changes other than the docstring.